### ISSUE TITLE
Figure out license

### ISSUE THREAD

**Issue by tobyweston** [xp-dojo/atdd-bank-account#7]
As for OSS, should we add copyright banners to all the files?

I'm keen to keep it open for people to use in classrooms but want to assert our rights at the same time

**Comment by lithgow** [comment]
Yeah, thinking about it, that probably makes sense. Any ideas on which license to go with? Something like copyleft?

**Comment by tobyweston** [comment]
I'm thinking Apache 2.0 📄

**Comment by lithgow** [comment]
Works for me. We can probably apply the same to the tdd kata, though we might want to link back to the original (which has no license) - https://github.com/sandromancuso/Bank-kata

**Comment by tobyweston** [comment]
@lithgow you happy with this? want to close the issue?

**Comment by lithgow** [comment]
Yup, closing

**Comment by tobyweston** [comment]
Bringing this up again, I propose we run a mixed licence model.

As I'd previously suggested I think we should use *Apache 2.0* for source and *Create Commons* (CC)(specific variant TBD) for documentation. This is fairly standard practice and I've spoken to the authors of the [Scala Cats](https://underscore.io/books/scala-with-cats/) book who use it for their stuff.

Have a look at he [License Finder](https://choosealicense.com/) if your interested.

I don't think this would be of concern to anyone's employer - assuming Apache is generally accepted, the CC is just a sidecar more appropriate for documentation (Apache is a *software* licence).

An example from this project:

> ## License
> The content of this project itself is licensed under the Creative Commons Attribution 3.0 Unported ? license, and the underlying source code used to format and display that content is licensed under the MIT license.

## Motivations

- [x] To retain our copyright (this is implicit and unrelated to licence choice, see #17 comments)
- [x] State terms and conditions with regards to limitations (protect us and our employers from misuse or liability issues)
- [x] Set out terms and conditions with regards to commercial use and distribution, monetization etc

## Remaining Issues

Assuming the above isn't contentious, we would still need to agree the CC variant. For example,  CC-BY-SA, CC-BY. or CC-BY-NC-SA This has an interesting side affect as it could affect any future revenue opportunities.

The choices are on the [Create Commons site](https://creativecommons.org/choose/)

"Six millions ways to die. Choose one":
- [ ] CC-BY
- [x] CC-BY-SA
- [ ] CC-BY-NC-SA
